### PR TITLE
worker-build: 0.0.17 -> 0.0.18

### PR DESCRIPTION
**Pull request by gmemstr**
## Description of changes

Somewhere along the way, a git dependency was added, meaning the cargoHash no longer worked. Also update us to 0.0.18: https://github.com/cloudflare/workers-rs/compare/v0.0.17...v0.0.18

Also the GitHub link was incorrect in the meta, which has been fixed.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Thank you for this, it looks like the build on linux-x86_64 is failing with a lock mismatch
https://github.com/NixOS/nixpkgs/pull/251932/checks?check_run_id=16266927802

File: pkgs/development/tools/worker-build/default.nix [hunk 1]
```diff
@@ -2,16 +2,21 @@
...
   pname = "worker-build";
-  version = "0.0.17";
+  version = "0.0.18";
 
   src = fetchFromGitHub {
     owner = "cloudflare";
     repo = "workers-rs";
     rev = "v${version}";
-    sha256 = "sha256-siKIE+RHAnZ8Lj3kWg7jEVo5t10dqc59OMrro26ClWo=";
+    sha256 = "sha256-z6m14IbMzgycwnQpA28e4taokDSVpfZOKIRmFIwLjbg=";
```
Comment (happysalada) [inline pkgs/development/tools/worker-build/default.nix:11]:
```suggestion
    hash = "sha256-z6m14IbMzgycwnQpA28e4taokDSVpfZOKIRmFIwLjbg=";
```

**Comment by gmemstr** [comment]
@happysalada Ah, found it - was missing a newline at the end of the committed Cargo.lock file so it didn't match 1:1. Fixed!

**Comment by happysalada** [comment]
Thank you!
if you use `nix-update worker-build`, it can generate the cargo.lock for you in case there needs to be one (so you never need to generate one manually). (just in case you weren't sure).

**Pull request merged**